### ISSUE TITLE
Multiple instantiated timers unsupported

### ISSUE THREAD

**Issue by JorySchossau** [ZhreShold/zupply#11]
If I make 3 timers, then only the first has reasonable values after calling reset on all 3, then a series of pause and resume on all 3. The other 2 have what look like uninitialized data (very large numbers). The timer api is pleasant to use, but only 1 timer can be instantiated.

**Comment by zhreshold** [comment]
Can you provide code and environment to reproduce that?
I can't reproduce the problem yet.

**Comment by JorySchossau** [comment]
Okay, I finally got an MWE to reproduce this! Curiously I find it required the timers to be used as data members of custom datatypes. What I wrote is a class that uses timers to time some of its own methods as if it were profiling itself (resume timer at method start, pause at method end). The code lets you specify at runtime how many times to call the methods before printing a summary of the time taken. At input or loop iteration of 1, all look fine. As the loop iteration grows (for example, to 10) then the zupply timers show wildly unreasonable times. Github repo for this code: www.github.com/JorySchossau/zupply-timers-test

```
$> time ./zt 1
predicted timer times
t1: 0
t2: 6
t3: 25

zupply timers
t1: 0 ms
t2: 6 ms
t3: 25 ms

real    0m0.035s
user    0m0.001s
sys     0m0.002s
```

But calling the profiled methods 10 times zupply starts imagining things:
```
$> time ./zt 10
predicted timer times
t1: 0
t2: 60
t3: 250

zupply timers
t1: 0 ms
t2: 6210 ms
t3: 25648 ms

real    0m0.317s
user    0m0.002s
sys     0m0.000s
```
